e2e/tests/accounts.feature

```diff
@@ -0,0 +1,78 @@
+
+  Scenario Outline: Using the Wrong Invocation for Accounts Namespace Commands
+	Given the user has a pocket client
+	When the user runs the command <mistaken_command>
+	Then the user should be able to see standard error containing "Error: "
+	And the user should be able to see standard error containing "Usage:"
+	And the user should be able to see standard error containing <command_template_start>
+	And the pocket client should have exited with error
+	Examples:
+	  | mistaken_command                                         | command_template_start       |
+	  | "accounts delete"                                        | "accounts delete"            |
+	  | "accounts export"                                        | "accounts export"            |
+	  | "accounts export-raw"                                    | "accounts export-raw"        |
+	  | "accounts import-armored"                                | "accounts import-armored"    |
+	  | "accounts import-raw"                                    | "accounts import-raw"        |
+	  | "accounts send-raw-tx"                                   | "accounts send-raw-tx"       |
+	  | "accounts send-tx"                                       | "accounts send-tx"           |
+	  | "accounts show"                                          | "accounts show"              |
+	  | "accounts sign"                                          | "accounts sign"              |
+	  | "accounts sign 7ab712998671b09e1a266ce6901000acb657833b" | "accounts sign"              |
+	  | "accounts sign message"                                  | "accounts sign"              |
+	  | "accounts update-passphrase"                             | "accounts update-passphrase" |
+
+  Scenario Outline: Running Commands With Correct Syntax and Bad Data
+	Given the user has a pocket client
+	Given the user runs the command <bad_command>
+	Then the user should be able to see standard output containing <message_contents>
+	And the pocket client should have exited without error
+	Examples:
+	  | bad_command                                                                   | message_contents               |
+	  | "accounts export thisisdefinitelyawrongaccount --pwd-decrypt=badpassword"     | "Address Error"                |
+	  | "accounts export-raw thisisdefinitelyawrongaccount --pwd-decrypt=badpassword" | "Address Error"                |
+	  | "accounts sign address testing"                                               | "invalid byte"                 |
+	  | "accounts create-multi-public thisisdefinitelyawrongaccount"                  | "error in public key creation" |
+
+  Scenario: Account Sign With Non-Existing Account
+	Given the user has a pocket client
+	When the user runs the command "accounts sign 7ab712998671b09e1a266ce6901000acb657833b 'example_message'"
+	Then the user should be able to see standard output containing "invalid byte"
+	And the pocket client should have exited without error
```

Comment:
Ideally, you'd want Then to be the last step, since its purpose is to observe outcomes. Here, each test is validating that the client exited without errors. That should be an individual scenario, not one to do in every scenario. For example:

Given the user has a pocket client
When the user runs the command ...
Then the user should see ...       <-- this is the outcome
---END---

Reply:
I understand where you're coming from; in this case -- a command line interface's API is comprised of its stdout, stderr and exit status. In particular, scripts often use the exit status to know whether the script can continue or needs to abort.

With regards to the `Then...` `And...`, the preference is aesthetic in this case; we're checking for two different outcomes, and it reads easier than `Then...` `Then...`.